kernel/src/osquery.cpp

```diff
@@ -62,6 +66,63 @@ static struct osquery {
   .major_number = OSQUERY_MAJOR
 };
 
+static int fileop_scope_callback(kauth_cred_t credential,
+                                 void *idata,
+                                 kauth_action_t action,
+                                 uintptr_t arg0,
+                                 uintptr_t arg1,
+                                 uintptr_t arg2,
+                                 uintptr_t arg3) {
+  vnode_t vp = (vnode_t)arg0;
+  if (action == KAUTH_FILEOP_EXEC && vp != NULL) {
+    // Someone is executing a file.
+    int path_len = MAXPATHLEN;
+
+    osquery_process_event_t *e =
+      (osquery_process_event_t *)osquery_cqueue_reserve(&osquery.cqueue,
+                                                        OSQUERY_PROCESS_EVENT);
+    if (e) {
```

Comment:
Could you return early if `e == NULL` and add a comment about the cases where a reservation fails?